tools/build_langserver/langserver/analyzer.go

```diff
@@ -139,32 +132,46 @@ func (a *Analyzer) builtInsRules() error {
 			if err != nil {
 				log.Warning("parsing failure: %s ", err)
 			}
-			// Iterate through the statement we got and add to statementMap
-			for _, statement := range stmts {
-				if statement.FuncDef != nil && !strings.HasPrefix(statement.FuncDef.Name, "_") {
-					content := string(asset)
-
-					ruleDef := newRuleDef(content, statement)
-					statementMap[statement.FuncDef.Name] = ruleDef
-
-					// Fill in attribute map if certain ruleDef is a attribute
-					if ruleDef.Object != "" {
-						if _, ok := attrMap[ruleDef.Object]; ok {
-							attrMap[ruleDef.Object] = append(attrMap[ruleDef.Object], ruleDef)
-						} else {
-							attrMap[ruleDef.Object] = []*RuleDef{ruleDef}
-						}
-					}
-				}
-			}
+
+			log.Info("Loading built-in build rules...")
+			a.loadBuiltinRules(stmts, string(asset))
 		}
 	}
 
-	a.BuiltIns = statementMap
-	a.Attributes = attrMap
+	for _, i := range a.State.Config.Parse.PreloadBuildDefs {
+		filePath := path.Join(core.RepoRoot, i)
+		bytecontent, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			log.Warning("parsing failure for preload build defs: %s ", err)
+		}
+		stmts, err := a.parser.ParseData(bytecontent, filePath)
+
+		log.Debug("Preloading build defs from %s...", i)
+		a.loadBuiltinRules(stmts, string(bytecontent))
+
+	}
 	return nil
 }
 
+func (a *Analyzer) loadBuiltinRules(stmts []*asp.Statement, fileContent string) {
+	for _, statement := range stmts {
+		if statement.FuncDef != nil && !strings.HasPrefix(statement.FuncDef.Name, "_") {
```

Comment:
can you not use `statement.FuncDef.IsPrivate` here?

Reply:
Oh yeah! I have that now right? Copy & paste... lol

Reply:
Oh no! the IsPrivate is only for args, I'm gonna add to funcDefs as well